rgd/geodata/models/mixins.py

```diff
@@ -22,16 +23,16 @@ class TaskEventMixin(object):
     should be registered as post_save events, not pre_save.
     """
 
-    task_funcs = None
-    """The task function."""
+    class Meta:
+        abstract = True
 
-    def _run_task(self):
-        if not isinstance(self.task_funcs, Iterable):
-            self.task_funcs = (self.task_funcs,)
-        if len(self.task_funcs) < 1:
+    status = models.CharField(max_length=20, default=Status.CREATED, choices=Status.choices)
+    task_funcs: List[Task] = []
+
+    def _run_tasks(self) -> None:
+        if not self.task_funcs:
             return
-        if not all([callable(f) for f in self.task_funcs]):
```

Comment:
FYI, as a matter of good code habits for whoever originally wrote this: `any/all([...])` should just be `any/all(...)` to allow for short-circuiting.

Reply:
Ah, thanks for the tip!